### ISSUE TITLE
JST globbing needs base single-star to prefix template fns

### ISSUE THREAD

**Issue by daemonburrito** [jazzband/django-pipeline#331]
It seems like, in order for `template_name()`, `base_path()`, and `compile_templates()` to work properly and name template functions with a directory name prefix, a single-star glob must be present as well as the double-star recursive one.

From the docs, this didn't work for me:

```
PIPELINE_JS = {
  'application': {
    'source_filenames': (
      'js/application.js',
      'js/templates/**/*.jst',
    ),
    'output_filename': 'js/application.js'
  }
}
```

This did:

```
PIPELINE_JS = {
  'application': {
    'source_filenames': (
      'js/application.js',
      'js/templates/*.jst',
      'js/templates/**/*.jst',
    ),
    'output_filename': 'js/application.js'
  }
}
```

tl;dr: If I didn't put in the `js/templates/*.jst`, I didn't get a directory name prefix on the template function name; e.g., `js/templates/foo/bar.jst` was just `bar()`, not `foo_bar()`.

I've seen the other issues re globbing, and I'm not exactly sure were my issue lands in relation to them. The OS is Ubuntu Precise (a vagrant box), so I don't think that's a problem (although the code does live in a virtualbox shared folder). It could also be that this is expected behavior? If so, I'd be happy to update the docs.
